subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractGradleceptionSmokeTest.groovy

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.smoketests
+
+import org.gradle.api.JavaVersion
+import org.gradle.api.specs.Spec
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+import org.gradle.integtests.fixtures.executer.GradleContextualExecuter
+import org.gradle.integtests.fixtures.executer.IntegrationTestBuildContext
+import org.gradle.internal.jvm.inspection.JvmInstallationMetadata
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+
+import java.text.SimpleDateFormat
+
+@Requires(value = TestPrecondition.JDK9_OR_LATER, adhoc = {
```

Comment:
Any idea why this test requires JDK9 or later? It seems like it searches its JDK on its own, even if the test is not running with JDK9.

Reply:
I'm not sure if both are required, but I didn't want to risk breaking CI...